### PR TITLE
jpa-specification

### PR DESCRIPTION
**Pull request by ihorromaniuk**
common practice to keep GET HTTP methods without body, so made POST instead

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -34,6 +35,11 @@ public ResponseEntity<?> getBookById(@PathVariable Long id) {
         return ResponseEntity.ok(bookService.getById(id));
     }
 
+    @PostMapping("/search")
+    public ResponseEntity<?> getBooksBySpec(@RequestBody BookParamsDto params) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]:
```suggestion
    public List<BookDto> getBooksBySpec(BookParamsDto params) {
```

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -34,6 +35,11 @@ public ResponseEntity<?> getBookById(@PathVariable Long id) {
...
+        return ResponseEntity.ok(bookService.findAll(params));
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java]:
```suggestion
        return bookService.findAll(params);
```

File: src/main/java/core/basesyntax/bookstore/dto/book/BookParamsDto.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package core.basesyntax.bookstore.dto.book;
+
+public record BookParamsDto(String partTitle, String[] authors) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/dto/book/BookParamsDto.java:3]:
```suggestion
public record BookParamsDto(String[] titles, String[] authors) {
```

File: src/main/java/core/basesyntax/bookstore/exception/BeanNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package core.basesyntax.bookstore.exception;
+
+public class BeanNotFoundException extends RuntimeException {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/bookstore/exception/BeanNotFoundException.java:3]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/core/basesyntax/bookstore/dto/book/BookParamsDto.java (hunk 3 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/dto/book/BookParamsDto.java:3]: discussed in chat

File: src/main/java/core/basesyntax/bookstore/exception/BeanNotFoundException.java (hunk 4 above)
Comment (ihorromaniuk) [reply src/main/java/core/basesyntax/bookstore/exception/BeanNotFoundException.java:3]: discussed in chat

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**